src/main/cpp/benchmarks/hash.cu

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+
+#include <cudf_test/column_utilities.hpp>
+
+#include <cudf/io/types.hpp>
+
+#include <hash.hpp>
+#include <nvbench/nvbench.cuh>
+
+constexpr auto min_width = 10;
+constexpr auto max_width = 10;
```

Comment:
Should this two be removed?

Reply:
The optimization effect is related to the schema, and I'm not sure how to write the benchmark for a good comparison.

Reply:
The benchmark should support running either (or both) lists and structs. You can switch benchmark types using a boolean flag. That flag should not be `constexpr` so we can compile all the code to verify syntax.
```
auto const bench_structs = false;
auto const data_profile = [&] {
  if (bench_structs) {
    ...
    .struct_types(...);
    return ...;
  } else {
    ...
    .list_type(...);
    return ...;
  }();
   
```